microbench/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoderBenchmark.java

```diff
@@ -0,0 +1,202 @@
+      nextBuf = Unpooled.wrappedBuffer(bodyPartBigBytes);
+    } else {
+      nextBuf = Unpooled.wrappedBuffer(intermediaryBytes);
+    }
+    DefaultHttpRequest req =
+        new DefaultHttpRequest(HttpVersion.HTTP_1_0, HttpMethod.POST, "/up");
+    req.headers().add(HttpHeaderNames.CONTENT_TYPE,
+                      "multipart/form-data; boundary=" + BOUNDARY);
+
+    long start = System.nanoTime();
+
+    DefaultHttpDataFactory defaultHttpDataFactory =
+        new DefaultHttpDataFactory(noDisk? 1024 * 1024 : 16 * 1024);
+    HttpPostRequestDecoder decoder =
+        new HttpPostRequestDecoder(defaultHttpDataFactory, req);
+    firstBuf.retain();
+    decoder.offer(new DefaultHttpContent(firstBuf));
+    firstBuf.release();
+    for (int i = 1; i < chunkNumber; i++) {
+      nextBuf.retain();
+      decoder.offer(new DefaultHttpContent(nextBuf));
+      nextBuf.release();
+      nextBuf.readerIndex(0);
+    }
+    finalBuf.retain();
+    decoder.offer(new DefaultLastHttpContent(finalBuf));
+    finalBuf.release();
+    while (decoder.hasNext()) {
+      InterfaceHttpData httpData = decoder.next();
+    }
+    while (finalBuf.refCnt() > 0) {
+      finalBuf.release();
+    }
+    while (nextBuf.refCnt() > 0) {
+      nextBuf.release();
+    }
+    while (finalBuf.refCnt() > 0) {
+      finalBuf.release();
+    }
+    long stop = System.nanoTime();
+    double time = (stop - start) / 1000000.0;
+    defaultHttpDataFactory.cleanAllHttpData();
+    defaultHttpDataFactory.cleanRequestHttpData(req);
+    decoder.destroy();
+    return time;
+  }
+
+  @Benchmark
+  public double multipartRequestDecoderHighDisabledLevel() {
+    final Level level = ResourceLeakDetector.getLevel();
+    try {
+      ResourceLeakDetector.setLevel(Level.DISABLED);
+      return testHighNumberChunks(false, true);
+    } finally {
+      ResourceLeakDetector.setLevel(level);
+    }
+  }
+
+  @Benchmark
+  public double multipartRequestDecoderBigDisabledLevel() {
+    final Level level = ResourceLeakDetector.getLevel();
+    try {
+      ResourceLeakDetector.setLevel(Level.DISABLED);
+      return testHighNumberChunks(true, true);
+    } finally {
+      ResourceLeakDetector.setLevel(level);
+    }
+  }
+
+  @Benchmark
+  public double multipartRequestDecoderHighSimpleLevel() {
+    final Level level = ResourceLeakDetector.getLevel();
+    try {
+      ResourceLeakDetector.setLevel(Level.SIMPLE);
+      return testHighNumberChunks(false, true);
+    } finally {
+      ResourceLeakDetector.setLevel(level);
+    }
+  }
+
+  @Benchmark
+  public double multipartRequestDecoderBigSimpleLevel() {
+    final Level level = ResourceLeakDetector.getLevel();
+    try {
+      ResourceLeakDetector.setLevel(Level.SIMPLE);
+      return testHighNumberChunks(true, true);
+    } finally {
+      ResourceLeakDetector.setLevel(level);
+    }
+  }
+
+  @Benchmark
+  public double multipartRequestDecoderHighAdvancedLevel() {
+    final Level level = ResourceLeakDetector.getLevel();
```

Comment:
nit: we use 4 spaces ... Please change everywhere to be consistent with our code-styling

Reply:
OK, I will double check.

Reply:
Done